provision-osx/tasks/configure_buildfarm_node.yml

```diff
@@ -3,42 +3,64 @@
 - include: configure_node_env_vars.yml
 
 -
-  block:
+  name: Set node key path
+  set_fact:
+    node_key: $HOME/.ssh_jenkins/jenkins_id_rsa
 
-  -
-    name: Create a directory to store key pair
-    file:
-      path: $HOME/.ssh_jenkins
-      state: directory
-      owner: "{{ ansible_ssh_user }}"
-      mode: 0700
+-
+  name: Create a directory to store key pair
+  file:
+    path: $HOME/.ssh_jenkins
+    state: directory
+    owner: "{{ ansible_ssh_user }}"
+    mode: 0700
+
+-
+  block:
   -
     name: Generate SSH key pair for Jenkins
     command: ssh-keygen -t rsa -f $HOME/.ssh_jenkins/jenkins_id_rsa -N "{{ credential_passphrase | default('') }}"
     args:
-      creates: "$HOME/.ssh_jenkins/jenkins_id_rsa"
+      creates: "{{ node_key }}"
+    when: (credential_private_key is undefined) and (credential_public_key is undefined)
+  when:
+    - (credential_private_key is undefined) or (credential_public_key is undefined)
```

Comment:
If I understood this code correctly, this code generates key pair on each macOS node, but in jenkins we only have one instance of credentials. So whichever node uploads credential last, will work and others won't.